web-ng/src/app/shared/models/feature.model.ts

```diff
@@ -15,33 +15,38 @@
  */
 
 import firebase from 'firebase/app';
+import { Map } from 'immutable';
 
 export interface Feature {
   readonly id: string;
   readonly layerId: string;
+  readonly properties?: Map<string, string | number>;
 }
 
 export class LocationFeature implements Feature {
   constructor(
     readonly id: string,
     readonly layerId: string,
     // TODO: User custom type instead of exposing types from data layer.
-    readonly location: firebase.firestore.GeoPoint
+    readonly location: firebase.firestore.GeoPoint,
```

Comment:
Leaky abstraction: GeoPoint is specific to the remote datastore layer. In our domain model it abstracted as `Point`

Reply:
We chatted; sorry I hadn't realized we don't have `Point` class defined in the model yet. Please disregard for this PR, we'll circle back.